core/src/main/java/com/arcrobotics/ftclib/util/FTCLibOpMode.java

```diff
@@ -0,0 +1,45 @@
+package com.arcrobotics.ftclib.util;
+
+import com.arcrobotics.ftclib.gamepad.GamepadEx;
+import com.qualcomm.hardware.lynx.LynxModule;
+import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
+import com.qualcomm.robotcore.eventloop.opmode.OpMode;
+
+import java.util.List;
+
+public abstract class FTCLibOpMode extends LinearOpMode {
+
+    protected GamepadEx driverOp, toolOp;
+
+    @Override
+    public final void runOpMode() throws InterruptedException {
+        driverOp = new GamepadEx(gamepad1);
+        toolOp = new GamepadEx(gamepad2);
+
+        initialize();
+
+        List<LynxModule> hubs = hardwareMap.getAll(LynxModule.class);
+        hubs.forEach(hub -> hub.setBulkCachingMode(LynxModule.BulkCachingMode.MANUAL));
+
+        waitForStart();
+
+        while (!isStopRequested() && opModeIsActive()) {
+            hubs.forEach(LynxModule::clearBulkCache);
+            run();
+            driverOp.readButtons();
+            toolOp.readButtons();
+            telemetry.update();
+        }
+    }
+
+    /**
+     * This method is like {@link OpMode#init()}
+     */
+    public abstract void initialize();
```

Comment:
Is there a reason for there being an `init` equivalent, but not `init_loop`? `init_loop` is very useful, especially for vision, and while here it can be done by putting a while loop at the end of the implementation for `initialize`, I feel like that goes against the style, especially since it isn't documented that `initialize` will not time you out for hanging the loop.

Reply:
Totally lapsed my mind. Definitely a feature that should be included.

Reply:
Let me know what you think of the recent change. @dansman805

Reply:
Recent changes look good